### ISSUE TITLE
Rule request: Detect dataclass() usage with Enum and its subclasses

### ISSUE THREAD

**Issue by correctmost** [astral-sh/ruff#15275]
### Motivation

There was a [bug in archinstall](https://github.com/archlinux/archinstall/issues/3055) because a `dataclass` decorator was incorrectly applied to an `Enum`:

```python
@dataclass
class Audio(Enum):
    NoAudio = 'No audio server'
    Pipewire = 'pipewire'
    ...
```

The [Python docs](https://docs.python.org/3/howto/enum.html#dataclass-support) warn about such usage, but I am not aware of any linter that currently catches the error:

> Adding dataclass() decorator to Enum and its subclasses is not supported. It will not raise any errors, but it will produce very strange results at runtime, such as members being equal to each other

Note: The docs mention that dataclass mixins can be used with enums, so there are some valid combinations of the constructs.

### Related discussions
- https://github.com/python/cpython/issues/114803
- https://github.com/python/cpython/pull/114891

### Documentation
- https://docs.python.org/3/howto/enum.html#dataclass-support

### Real world bugs
- https://github.com/archlinux/archinstall/pull/3058

**Comment by MichaReiser** [comment]
Thanks, this makes sense to me
